Yoast/YoastSEO.js
# Refactor topic distribution to include morphology

## Summary

This PR can be summarized in the following changelog entry:

* Changes the scoring schema as follows:
For every sentence determine, whether it reflects the topic of the text (based on the keyphrase and synonym information) or not. The keyphrase or a synonym phrase are taken to be reflected in the sentence, if [ for a keyphrase or a synonym phrase of 3 words or shorter ] all content words appear in this sentence or [ for a keyphrase or a synonym phrase of 4 words or longer ] if at least three content words appear in this sentence and all other content words appear in one sentence before of after the sentence in question.
For every sentence take the maximal score that it received when being checked for reflecting the topic based on the keyphrase and all synonym phrases.
Take one third of the text (in sentences) and calculate an average score. 
Move down by one sentence and calculate an average again. Continue until the end of the text while maintaining the window size.
Take the results of the per-window analysis and calculate Gini coefficient based on them.
Use the resulting Gini coefficient as a metric of uniformity of the distribution to return specific feedback strings to the user.

## Relevant technical choices:

*  Markers: marks all sentences that contain no topic forms. Markers are only enabled when the result is bad or okay.

## Test instructions

This PR can be tested by following these steps:

* Add a text with more than 200 words and a keyword.
* As long as your keyword doesn't occur in the text, you should get a gray bullet and a prompt to include your keyword in the text.
* Include your keyword in the text once: you should get a red bullet. When using the marker, all sentences except the sentence with the keyword should be marked.
* Add the keywords more often in your text. At some point you should get an orange bullet. The marker should still work.
* Add some more keywords. You should get a green bullet.
* Repeat the same steps with a keyphrase consisting out of 2-3 words and a keyphrase consisting of 4+ words and make sure you get meaningful results.

Fixes #1755
